Issue 2077: Add sample playwright tests for folium

hansthen opened the pull request:
Gentlemen, I created a sample test using playwright and streamlit. This is not a full test suite, but more of a demonstration how we can use Playwright to test our example apps. I have used Playwright for a few other projects and I like how it works. It also comes with a nice `codegen` utility that can be used to jumpstart creating tests. 

Let me know what you think of it.

hansthen committed:
Add sample playwright tests for folium

First try: just do the draw plugin.

hansthen committed:
Undo module import

Instead do dynamic import as a workaround

hansthen committed:
Add dev requirements

hansthen committed:
Move pixelmatch away from conda and mamba

hansthen committed:
Enable static file serving

hansthen committed:
Upload screenshots after failure

hansthen committed:
There are some small differences in the tiles

hansthen commented:
I am okay with the results for now. Please let me know what you think.

ocefpaf commented:
> I am okay with the results for now. Please let me know what you think.

I don't know that `playwright` and I'm not familiar with streamlit enough to review this PR. I'll leave this one for you and Frank. Sorry.

hansthen committed:
New snapshot code

hansthen commented:
> > I am okay with the results for now. Please let me know what you think.
> 
> I don't know what `playwright` is, and I'm not familiar enough with `streamlit` to review this PR. I'll leave this one for you and Frank. Sorry.

No problem. Playwright is a UI testing framework that is also used by the streamlit-folium people. It feels a lot like selenium, but with less baggage. I will make this PR a draft for now and focus first on getting the regressions to work.

ocefpaf commented:
> It feels a lot like selenium, but with less baggage.

Cool! I'll watch this closely to learn something new.

hansthen commented:
> > It feels a lot like selenium, but with less baggage.
> 
> Cool! I'll watch this closely to learn something new.

Thank you for your help! Half a dozen issues closed in a single day. I will prepare a release soon.

hansthen committed:
Second iteration of the tests

hansthen committed:
Fix merge conflict

hansthen committed:
Moved snapshots to new location

Also created new screenshots due to changing tile images

hansthen committed:
Include pixelmatch dependency

hansthen committed:
Create visual snapshot testing

hansthen committed:
Small fixes

hansthen committed:
Add test file

hansthen committed:
New screenshots

hansthen committed:
New screenshots

hansthen committed:
Remove snapshot tests regular workflow

hansthen committed:
Remove snapshot tests

hansthen committed:
Cleanup playwright tests

hansthen committed:
Improvements

hansthen committed:
New screenshots

Removed the tiles, because there was a difference in between
github and local.

Removed the playwright tests.

hansthen committed:
Add a tests to save all package versions

hansthen committed:
Try again

hansthen committed:
Check versions

hansthen committed:
Aargh

hansthen committed:
Also save the generated html

hansthen committed:
Also upload the generated maps

for easier comparison

hansthen committed:
Check the chromium version

hansthen committed:
Used remotely created screenshots

hansthen committed:
Rerun 5 times
